lib/file_transfer_agent/encrypt_util.js

```diff
@@ -125,206 +125,268 @@ function EncryptUtil(encrypt, filestream, temp) {
     return newMatDesc;
   }
 
+  function createEncryptionMetadata(encryptionMaterial, keySize, encryptedKey, dataIv, keyIv = null, dataAad = null, keyAad = null) {
+    const matDesc = MaterialDescriptor(
+      encryptionMaterial.smkId,
+      encryptionMaterial.queryId,
+      keySize * 8
+    );
+
+    return new EncryptionMetadata(
+      encryptedKey.toString(BASE64),
+      dataIv.toString(BASE64),
+      matDescToUnicode(matDesc),
+      keyIv,
```

Comment:
So for me it is a bit strange - data iv is a byte array and key iv is already encoded?

Reply:
All IVs are now base64 encoded. Check lines above